internal/dag/builder.go

```diff
@@ -518,6 +521,18 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 			svhost.Secret = sec
 			svhost.MinProtoVersion = annotation.MinProtoVersion(proxy.Spec.VirtualHost.TLS.MinimumProtocolVersion)
 
+			if proxy.Spec.VirtualHost.TLS.EnableFallbackCertificate && !isBlank(b.FallbackCertificate) {
+				secretName := splitSecret(b.FallbackCertificate, "projectcontour")
```

Comment:
What should the default namespace for this fallback secret be? I don't think we should hard-code it here, but if it isn't in a namespace we are already watching, then we won't be able to look it up.

Maybe we should *require* the namespace field on the fallback secret name, to ensure that only one is ever used? It's not clear to me what the right default namespace should be here.

We should also check the secret name flag on startup and ensure that it is consistent with the root namespaces handling of Secrets?

Reply:
I think we can do it with a string that *may* include a namespace name `<namespace>/secretname` or `secretname` would both be valid. Agreed that the secret name flag should be checked on startup if we do this, to ensure that the namespace is one we are watching for Secrets. We may also need to add it if we aren't watching already.

Reply:
I was changing this around a bit today (just haven't pushed). My intent was to make the namespace/name what was required. By doing that I can check that 1. users followed the right pattern, and 2. remove the need for a default namespace later on. 

If we allow just a `name`, then it would have to be in the context of where Contour is deployed and we could use the downward API logic that already exists to use as the default.

Reply:
What should the namespace be if it secret name isn't fully qualified?

Reply:
> I was changing this around a bit today (just haven't pushed). My intent was to make the namespace/name what was required. By doing that I can check that 1. users followed the right pattern, and 2. remove the need for a default namespace later on.

Yup, I agree that requiring a fully qualified name makes sense here.

Reply:
I'm okay with requiring a namespace, it would be a little complicated to explain how it interacted with the downward API bits.

Reply:
Right now I have it as a `namespace/name` and Contour checks for that on startup. This is passed in via an arg on `contour serve` and in the config file. 

One way to maybe make this more clear would be to remove the ability to set the command like arg, and only use the config file. The file could extend another struct to have `fallbackCertificate` and then ask for the name and namespace in separate fields. 

Example:

```yaml
tls:
    minimum-protocol-version: "1.1"
    fallbackCertificate:
      name: "secret"
      namespace: "userns"
```

Reply:
I like this. I'd like to move away from adding cmdline flags, and start pushing people towards using the config file only.